repository/shift_repository.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from flask import jsonify
+
+from datetime import datetime, timezone
+from exception import EventNotFoundError, InvalidArgumentError
+from domain import session_scope, ShiftRequestVolunteer, ShiftRequest
+
+
+class ShiftRepository:
+
+    def __init__(self):
+        pass
+
+    def get_shift(self, userId):
+        """
+            Retrieves all shift events for a given user that have not ended yet.
+
+            :param userId: ID of the user whose shifts are being queried.
+            :return: A list of shift records or an empty list if none found.
+        """
+        now = datetime.now()
+        with session_scope() as session:
+            try:
+                # only show the shift that is end in the future
+                shifts = session.query(ShiftRequestVolunteer).join(ShiftRequest).filter(
+                        ShiftRequestVolunteer.user_id == userId,
+                        ShiftRequest.endTime > now).all()
+                # check if there's some results
+                if not shifts:
+                    logging.info(f"No active shifts found for user {userId}")
+                    return []
+                shift_records = []
+                # write shift information into list
+                for shift in shifts:
+                    shift_record = {
```

Comment:
Please create a dataclass (look at [the docs](https://docs.python.org/3/library/dataclasses.html) for an example) with typing to represent this mapped model. I would put the new file in domain/types. 

The dataclass library isn't actually installed right now, so before working on this, run the following command (in the root of the project):

```bash
pipenv install dataclasses
pipenv lock
pipenv requirements > requirements.txt
```

Reply:
I have create a dataclass in domain/types and use this dataclass in get_shift function. If there's something wrong with it, please feel free to contact me.